### DISCUSSION TITLE
Problems in introducing third-party class libraries

### DISCUSSION THREAD

**Discussion by hanyibo520** [theos/theos#675] [Questions]
### What are the steps to reproduce this issue?

I added three third-party frameworks to my project（CocoaHTTPServer、CocoaAsyncSocket，CocoaLumberjack），I added it to / opt / theos / vendor / lib,
<img width="1299" alt="截屏2022-04-19 下午7 57 25" src="https://user-images.githubusercontent.com/20529339/163998761-f780348b-abb5-42e2-81ee-e4a6450cfbe4.png">
makefile：
<img width="544" alt="截屏2022-04-19 下午8 02 38" src="https://user-images.githubusercontent.com/20529339/163999030-0f3886d4-9d0a-4edb-a380-cd5cbd2293a7.png">
but When I compile：
<img width="1437" alt="截屏2022-04-19 下午8 03 50" src="https://user-images.githubusercontent.com/20529339/163999190-dc938253-5c21-42f8-af2e-71cad699d4b0.png">
Tweak injection failed.Excuse me, why is this?

### What happens?

The third-party class library was added as required, but the compilation failed。

### What were you expecting to happen?

It can be compiled and injected successfully.

### What OS are you seeing the problem on?

macOS

### What OS version does it have installed?

macOS 11.4 M1 Mac

### What toolchain and version are you using?

Xcode 13.0

### Which SDK version are you using?

iOS 9.3

### What OS is your client device running?

iOS

### What OS version does it have installed?

iOS 13.6

### Which device model is it?

iPhone 6s

### Which jailbreak is it using (if relevant)?

checkra1n

**Comment by L1ghtmann** [comment]
Three things:
- User-added libraries and frameworks should be placed in $THEOS/lib, not $THEOS/vendor/lib
- The frameworks you've added and are linking against are built for, at minimum, iOS 8/9 and your target appears to still be set to the default which is iOS 7.
- The frameworks you've added and are linking against aren't built for armv7s, which is an architecture you're currently building for. 

The solution to the latter two issues is to change the relevant variables in your Makefile. 

You'll want to change your ARCHS to be just `arm64` and your TARGET to be `iphone:clang:your_target_sdk:newer_min_target_version`.

If you're building for just your device, the only arch you need to build for is arm64. If you're building for public release, however, you'll want to also build for arm64e. Note that, based on the age of the frameworks you're linking against, they may not include an arm64e slice in which case you'll have to recompile them for arm64e or find newer alternatives.

**Comment by hanyibo520** [comment]
First of all, thank you!!
I did it according to your requirements. Tweak has been compiled, but when I opened the app to hook, I found that the injection failed.
<img width="645" alt="截屏2022-04-20 下午12 17 11" src="https://user-images.githubusercontent.com/20529339/164149478-e57a32bd-6700-419a-9129-d3d156544deb.png">
<img width="400" alt="截屏2022-04-20 下午12 27 29" src="https://user-images.githubusercontent.com/20529339/164150451-7091c460-b794-45b3-9409-ff7ba6075bc8.png">
<img width="1024" alt="截屏2022-04-20 下午12 19 20" src="https://user-images.githubusercontent.com/20529339/164149776-04bfcda4-8178-4964-b1fd-af9417c96d21.png">
